src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -32,25 +32,31 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         final SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
-        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
-                .filter(h -> h.getValue().getHearingDateTime().isAfter(LocalDateTime.now()))
-                .distinct()
-                .findFirst();
-
         final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
 
         if (HearingRoute.GAPS.equals(sscsCaseData.getRegionalProcessingCenter().getHearingRoute())) {
             response.addError(POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS);
+            return response;
         }
 
+        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
+                .filter(h -> h.getValue().getStart().isAfter(LocalDateTime.now()))
+                .distinct()
+                .findFirst();
+
         optionalHearing.ifPresentOrElse(hearing -> setPostponementRequest(hearing, sscsCaseData),
                 () -> response.addError("There are no hearing to postpone"));
 
         return response;
     }
 
     private void setPostponementRequest(Hearing hearing, SscsCaseData sscsCaseData) {
```

Comment:
Formatting seems to have gone a bit awry in this method.

Reply:
If you mean the indentation, that's what the IDE proposed.